lib/tpm2_errata.c

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2017, Alibaba Group
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// 3. Neither the name of Intel Corporation nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+#include <stdarg.h>
+#include <stdbool.h>
+
+#include "log.h"
+#include "tpm2_errata.h"
+#include "tpm2_util.h"
+
+struct tpm2_errata_desc {
+    UINT32 spec_rev;
+    UINT32 errata_ver;
+    void (*fixup) (struct tpm2_errata_desc *errata, va_list *ap);
+    const char *name;
+};
+
+/* Published errata information */
+struct tpm2_errata_info {
+    UINT32 spec_rev;
+    UINT32 errata_ver;
+    UINT32 year;
+    UINT32 day_of_year;
+} known_errata_info[] = {
+    /* Specification Revision 1.16 October 30, 2014 */
+    { 116, 000, 2014, 303 },
+    /* Errata Version 1.2 February 16, 2015 */
+    { 116, 120, 2015,   4 },
+    /* Errata Version 1.3 June 16, 2015 */
+    { 116, 130, 2015, 167 },
+    /* Errata Version 1.4 January 15, 2016 */
+    { 116, 140, 2016,  15 },
+    /* Errata Version 1.5 September 21, 2016 */
+    { 116, 150, 2016, 265 },
+    /* Specification Revision 1.38 September 29, 2016 */
+    { 138, 000, 2016, 273 },
+    /* Errata Version 1.0 January 16, 2017 */
+    { 138, 100, 2017,  16 },
+    /* Errata Version 1.1 March 2, 2017 */
+    { 138, 110, 2017,  61 },
+}, *this_errata_info;
+
+static void
+fixup_sign_decrypt_attribute_encoding(struct tpm2_errata_desc *errata,
+                                      va_list *ap) {
+
+    TPMA_OBJECT *attrs = va_arg(*ap, TPMA_OBJECT *);
+printf("applied\n");
```

Comment:
get rid of this, if you want any verbose information their is macros in log.h like the one you used below.